### ISSUE TITLE
eslint-plugin-tsdoc: allow policies to be ignored in tsdoc/syntax rule

### ISSUE THREAD

**Issue by bitjson** [microsoft/tsdoc#220]
Now that less-strict parsing for `@param` tags is merged (https://github.com/microsoft/tsdoc/issues/128), I'd love to be able to loosen the requirements of the `tsdoc/syntax` rule to allow for certain deviations from the expected syntax.

Primarily, VSCode's [built-in JSDoc support](https://code.visualstudio.com/docs/languages/javascript#_jsdoc-support) automatically adds a doc comment where each `@param` tag is followed immediately by a description, without the expected hyphen. With the following `.eslintrc` configuration:

```js
{
  rules: {
    'tsdoc/syntax': 'warn'
  }
}
```

This snippet is valid:
```ts
/**
 * Description
 * @param a - first number
 * @param b - second number
 */
const add = (a: number, b: number) => a + b;
```

While this snippet:
```ts
/**
 * Description
 * @param a first number
 * @param b second number
 */
const add = (a: number, b: number) => a + b;
```

Warns with the following:

```
tsdoc-param-tag-missing-hyphen: The @param block should be followed by a parameter name and then a hyphen eslint(tsdoc/syntax)
```

I'd love to be able to ignore this particular syntax deviation with something like:
```js
{
  rules: {
    'tsdoc/syntax': ['warn', ignore: ['tsdoc-param-tag-missing-hyphen'] ]
  }
}
```

**Comment by octogonz** [comment]
This is a good idea. However I think we might want to approach it via the [tsdoc.json config file](https://github.com/microsoft/tsdoc/tree/master/tsdoc-config) rather than via ESLint's configuration.  The reason is that you are not really suppressing a warning, but rather defining the kind of TSDoc parsing that you want for your source files.  Putting the suppression in **tsdoc.json** will prevent other tools from reporting the same errors when they invoke the parser.

Long term, we intend to provide a "lax mode" as well as other compatibility settings for the TSDoc parser.  But in the short term, this kind of problem can be solved pretty well by simply filtering out specific error IDs as you suggest.  (The reason that's not a *perfect* solution is that the syntax highlighting may sometimes color the bad characters in red even though the error message is hidden.)

**eslint-plugin-tsdoc** already supports **tsdoc.json**, so we would simply need to improve [tsdoc.schema.json](https://github.com/microsoft/tsdoc/blob/master/tsdoc/schemas/tsdoc.schema.json) to support something like this:

**your-project/tsdoc.json**
```ts
{
  "$schema": "https://developer.microsoft.com/json-schemas/tsdoc/v0/tsdoc.schema.json",
  "ignoreMessageIds": [
    "tsdoc-config-file-not-found"
  ]
}
```
...and then pass that setting through to the parser.  It's mostly plumbing work, since in the end all we're doing is adding a filter to [ParserMessageLog.addMessage()](https://github.com/microsoft/tsdoc/blob/4b8311f30f80265be255eca7cb30829ee2ba9296/tsdoc/src/parser/ParserMessageLog.ts#L24).

**Comment by zoechi** [comment]
I just run into a problem and disabling rules doesn't work as usual where I have a section in the `.eslintrc.json` for `*spec.ts` files

https://jestjs.io/docs/en/configuration#testenvironment-string

```
/**
 * @jest-environment jsdom
 */
```
```
  2:4  warning  tsdoc-characters-after-block-tag: The token "@jest" looks like a TSDoc tag but contains an invalid character "-"; if it is not a tag, use a backslash to escape the "@"  tsdoc/syntax
```
but haven't found a way to disable this rule.

**Comment by countzero** [comment]
I would second this request. Our use case is the need to disable the ts-doc-at-sign-in-word rule, The following should not trigger any warnings:

```TypeScript
/**
 * A method description containing an example@emailaddress.com causing a linter warning.
 */
private getEmailAddress (): string {
    return 'example@emailaddress.com';
}
```

**Comment by Rob4226** [comment]
Yes, please an `"ignoreMessageIds"` is very much needed.  Unless I am missing something it seems the eslint-plugin-tsdoc is all or nothing.  Is there any way to stop reporting on some message ids at the moment?

**Comment by ssbarnea** [comment]
So basically there is no way to keep using `@author` if we want to use `eslint-plugin-tsdoc`?

**Comment by jkowalleck** [comment]
re: https://github.com/microsoft/tsdoc/issues/220#issuecomment-1369968157
> So basically there is no way to keep using `@author` if we want to use `eslint-plugin-tsdoc`?

of cause you could create own tags.

this is my `tsdoc.json` file:
```json
{
  "$schema": "https://developer.microsoft.com/json-schemas/tsdoc/v0/tsdoc.schema.json",
  "tagDefinitions": [
    {
      "tagName": "@since",
      "syntaxKind": "block",
      "allowMultiple": false
    },
    {
      "tagName": "@author",
      "syntaxKind": "block",
      "allowMultiple": true
    },
    {
      "tagName": "@TODO",
      "syntaxKind": "block",
      "allowMultiple": true
    }
  ]
}
```

**Comment by Patrick-clone** [comment]
This would be a great (and essential) addition.  What is the status of this?

**Comment by GitMurf** [comment]
> eslint-plugin-tsdoc already supports tsdoc.json, so we would simply need to improve [tsdoc.schema.json](https://github.com/microsoft/tsdoc/blob/master/tsdoc/schemas/tsdoc.schema.json) to support something like this:

@octogonz did anything ever come of this? I liked your proposed approach. Are there any solutions today? Thanks.
